tensorflow/python/ops/image_ops.py

```diff
@@ -252,13 +252,12 @@ def _slice_channels(image):
     list of tensor, one for each channel.
   """
 
-  image_shape = image.get_shape()
-  image_rank = len(image_shape)
+  image_rank = array_ops.rank(image)
 
   begin = [0] * (image_rank - 1)
   size = [-1] * (image_rank - 1) + [1]
 
-  channels = [array_ops.slice(image, begin + [i], size) for i in range(image_shape[-1])]
+  channels = [array_ops.slice(image, begin + [i], size) for i in range(3)]
```

Comment:
I was looking for this! Turns out it didn't exist (with the axis argument) in r0.9, but very nice that it was added in the master branch.